resources/aws_ec2.go

```diff
@@ -708,3 +750,50 @@ func (obj *AwsEc2Res) snsPostHandler(w http.ResponseWriter, req *http.Request) {
 		log.Printf("%s: Post: %s", obj, string(post))
 	}
 }
+
+// snsMakeTopic creates a topic on aws sns.
+func (obj *AwsEc2Res) snsMakeTopic(snsClient *sns.SNS) (*string, error) {
+	// make topic
+	topicInput := &sns.CreateTopicInput{
+		Name: aws.String(SnsTopicName),
+	}
+	topic, err := snsClient.CreateTopic(topicInput)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("%s: Created SNS Topic", obj)
+	return topic.TopicArn, nil
+}
+
+// snsDeleteTopic deletes the sns topic.
+func (obj *AwsEc2Res) snsDeleteTopic(snsClient *sns.SNS) error {
+	// delete the topic
+	dtInput := &sns.DeleteTopicInput{
+		TopicArn: obj.snsTopicArn,
+	}
+	log.Printf("%s: Deleting Topic", obj)
+	if _, err := snsClient.DeleteTopic(dtInput); err != nil {
+		return errwrap.Wrapf(err, "error deleting topic")
+	}
+	return nil
+}
+
+// Close cleans up when we're done. This is needed to delete some of the AWS
+// objects created for the SNS endpoint.
+func (obj *AwsEc2Res) Close() error {
+	var errList error
+	if obj.WatchListenAddr != "" {
+		// make sns client
+		snsClient, err := obj.snsClient()
```

Comment:
Why didn't we just re-use the client from Init() ? You took that out... Maybe I did not explain clearly. Better explanation here: https://github.com/purpleidea/mgmt/pull/241#discussion_r150230274